test/common/http/header_utility_test.cc

```diff
@@ -289,6 +289,21 @@ contains_match: somevalueinside
   EXPECT_EQ("somevalueinside", header_data.value_);
 }
 
+TEST(HeaderDataConstructorTest, StringMatchSpecifier) {
```

Comment:
Why aren't some of the tests in this file now not `DEPRECATED_FEATURE_TEST`?

Reply:
Updated to use `DEPRECATED_FEATURE_TEST`.